lua/nvim-treesitter/locals.lua

```diff
@@ -10,14 +10,14 @@ local M = {
   locals={}
 }
 
-function M.is_supported(lang)
+local function is_supported(lang)
   return queries.get_query(lang, "locals") ~= nil
 end
 
 function M.collect_locals(bufnr)
   local ft = api.nvim_buf_get_option(bufnr, "ft")
 
-  if not ft then return end
+  if not (ft and is_supported(ft)) then return end
 
   local query = queries.get_query(ft, 'locals')
```

Comment:
is_supported calls get_query, and you call get_query right after. Would it be better to check if query is not nil after the call to get_query ?
That's what i usually do to avoid repeating function calls.
The only use case for this function would be if it was called in multiple other places, but right now it is not.
Also we should follow neovim naming conventions, keep `ft` as `ft` and not `lang`. I'll try to bear that in mind in my work too.